internal/dockercompose/state.go

```diff
@@ -101,6 +104,34 @@ func (s State) HasEverBeenReadyOrSucceeded() bool {
 	return !s.LastReadyTime.IsZero()
 }
 
+func populateContainerStatus(state *types.ContainerState) string {
+	if state.Health == nil {
+		return state.Status
+	}
+	if state.Health.Status == types.Starting {
+		return ContainerStatusRestarting
```

Comment:
i wonder if this should be created?

Reply:
This, combined with a change in `RuntimeStatus()` method, improves the user experience on container rebuild in case of docker containers with healthcheck. It reflects that, there is additional "pending" state of healthcheck status.

Without it, in case when the healthcheck started failing after the rebuild, the runtime status would be first green and only after the healthcheck is considered as failing by docker, the runtime status would become red.

With this change the runtime status will be "pending" as long as the the healthcheck is "healthcheck: starting".

By the runtime status I mean the red element on Tilt UI:
![image](https://github.com/user-attachments/assets/67f4ddd5-b6ed-43aa-bf79-70d0095333e8)

Reply:
do ContainerStatusRestarting and ContainerStatusCreated have different displays in the UI?

Reply:
`ContainerStatusCreated` **->> green runtime status**
![image](https://github.com/user-attachments/assets/bb4eb828-9448-4a23-bf1d-90d830b51522)

`ContainerStatusRestarting ` **->> below animated. At the end of animation is a transition to unhealthy state.**

https://github.com/user-attachments/assets/6371d470-11cf-40f4-b541-a00fdd71d619

Again, without the change in question when the healthcheck procedure is ongoing the runtime status would mistakenly indicate green status.